unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneResourcesLoadTracker/Tests/MockedSharedComponent.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using DCL.Components;
+using DCL.Controllers;
+using DCL.Models;
+
+namespace Tests
+{
+    public class MockedSharedComponent : ISharedComponent
+    {
+        private event Action<ISharedComponent> OnReady;
+        public string id;
```

Comment:
We should use the `Substitute.For` instead of creating a new class

Reply:
I don't think I can.
Need to mock that component is ready and I don't think that there is a way to do that using NSubstitute

Reply:
```
bool ready = false;
componentMock.WhenForAnyArgs(x => x.CallWhenReady(Args.Any<Action<ISharedComponent>>)).Do(x => ready = true);

yield return new WaitUntil(() => ready);
```

Written from the top of my head, it probably has some issues, havent checked in the IDE.
It's also fairly similar to what it's done in: `LoadWrapperShould.UnloadWhenEntityDestroyedBeforeFinishing`.

Reply:
I remember doing it for biw but I cannot find where I did it. You can raise events 
https://nsubstitute.github.io/help/raising-events/

Reply:
I think we both missed the point of the test.

This one should do what you are looking for, Pato:
```csharp
List<Action<ISharedComponent>> callbacks = new List<Action<ISharedComponent>>();
        var componentMock = Substitute.For<ISharedComponent>();
        componentMock.WhenForAnyArgs(x => x.CallWhenReady(Args.Any<Action<ISharedComponent>>)).Do(info =>
        {
            callbacks.Add(ArgAt<Action<ISharedComponent>>(0)); 
        });
```

In that `callbacks` list you have everything subscribed to `CallWhenReady` so you can manually invoke them.